_posts/2020-03-02-ios-deploys-super-weird-totally-normal.markdown

```diff
@@ -0,0 +1,110 @@
+
+Today I want to explore the differences between deploying iOS software and front-end/back-end web software. Some of
+these differences are inherent to how the code gets executed, and some of the differences are incidental to choices
+that Apple has made. These are constraints that iOS developers need to work within. As Artsy has adopted React
+Native over the pats four years, we have had more and more of our web engineering colleagues contributing to our
+iOS app. For these web engineers, getting familiar with the iOS deploy constraints is as important as getting to
+know Xcode and CocoaPods.
+
+We're going to use a case study to frame today's discussion. Artsy's Mobile Experience team recently got a ticket
+from our Platform team. The nature of the ticket itself doesn't matter, but it involved a change we were making to
+our API. I fixed the bug and submitted a pull request, which quickly got merged. We tested the bug fix in our next
+beta deploy, and everything looked fine. Great so far.
+
+Artsy releases iOS software on a two-week release cadence, and this bug fix was scheduled to be released to users
+the following week. But our back-end team wanted to quickly deploy that API change, which would require the app bug
+fix to get released to users _first_.
+
+So... what to do?
+
+Let's actually pause for a moment and consider one of the assumptions in that last paragraph. If you're a web
+engineer, the idea of releasing only every two weeks might seem pretty strange! I mean, why not release
+continuously? For example, Artsy's website gets deployed to our staging environment after every merged pull
+request, and staging then gets promoted to production several times a day. This process is generally referred to as
+"continuous delivery", and
+[it has a lot of advantages](https://www.thoughtworks.com/insights/blog/case-continuous-delivery). That's why it's
+so common among web engineering teams. So why not use continuous delivery on iOS apps?
+
+There are two reasons we can't use continuous delivery on iOS. First, continuous delivery is only really possible
+when you control where the software gets executed (or, in the case of web front-ends, where the client-side code
+gets served from). Artsy controls our own servers, so we can delivery web software continuously. However, the Artsy
+iOS app runs on our users' devices, instead. We can't push out updates to users' iPhones or iPads in the same way
+we can push updates to our servers.
+
+iOS apps are binary executables that are distributed through Apple's App Store, and updates to apps have to be
+pulled down by devices. Even if _most_ users have automatic updates turned on, those updates are typically
+installed overnight. Consequently, there's quite a lag between when we deploy an iOS update and when users run the
+code. While it only takes about a week for 80% of our users to update to the latest version, there's a very long
+tail after that.
+
+// TODO: Insert image on Monday.
```

Comment:
On Monday, I'll grab an image like this from Looker that demonstrates the long tail described above:

<img width="1210" alt="Screen Shot 2020-02-28 at 13 51 43" src="https://user-images.githubusercontent.com/498212/75581804-40336580-5a38-11ea-8939-e2ae1dd98fc7.png">

Waiting for Monday so that we can see the new update rolled out a bit more. Since this is % and not a total # of sessions, there's no propriety data here (but I checked with Data just in case). @mikehrom suggested better colours, which also sounds like a good idea.